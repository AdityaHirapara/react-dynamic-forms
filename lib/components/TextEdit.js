"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _revalidator = require("revalidator");

var _underscore = require("underscore");

var _underscore2 = _interopRequireDefault(_underscore);

var _stringHash = require("string-hash");

var _stringHash2 = _interopRequireDefault(_stringHash);

var _formGroup = require("../formGroup");

var _formGroup2 = _interopRequireDefault(_formGroup);

require("./css/textedit.css");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                *  Copyright (c) 2015, The Regents of the University of California,
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                *  through Lawrence Berkeley National Laboratory (subject to receipt
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                *  of any required approvals from the U.S. Dept. of Energy).
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                *  All rights reserved.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                *  This source code is licensed under the BSD-style license found in the
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                *  LICENSE file in the root directory of this source tree.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */

/**
 * Form control to edit a text field.
 * Set the initial value with 'initialValue' and set a callback for
 * value changed with 'onChange'.
 */
var TextEdit = function (_React$Component) {
  _inherits(TextEdit, _React$Component);

  function TextEdit(props) {
    _classCallCheck(this, TextEdit);

    var _this = _possibleConstructorReturn(this, (TextEdit.__proto__ || Object.getPrototypeOf(TextEdit)).call(this, props));

    _this.state = {
      value: props.value,
      error: null,
      errorMsg: "",
      missing: false,
      type: "text"
    };
    return _this;
  }

  _createClass(TextEdit, [{
    key: "isEmpty",
    value: function isEmpty(value) {
      return _underscore2.default.isNull(value) || _underscore2.default.isUndefined(value) || value === "";
    }
  }, {
    key: "isMissing",
    value: function isMissing(v) {
      return this.props.required && !this.props.disabled && this.isEmpty(v);
    }
  }, {
    key: "getError",
    value: function getError(value) {
      var result = { validationError: false, validationErrorMessage: null };

      // If the user has a field blank then that is never an error. Likewise if the field
      // is disabled then that is never an error.
      if (this.isEmpty(value) || this.props.disabled) {
        return result;
      }

      // Validate the value with Revalidator, given the rules in this.props.rules
      var obj = {};
      obj[this.props.name] = value;

      var properties = {};
      properties[this.props.name] = this.props.validation;

      var rules = this.props.validation ? { properties: properties } : null;
      if (obj && rules) {
        var validation = (0, _revalidator.validate)(obj, rules, { cast: true });
        var name = this.props.name || "Value";

        var msg = void 0;
        if (!validation.valid) {
          msg = name + " " + validation.errors[0].message;
          result.validationError = true;
          result.validationErrorMessage = msg;
        }
      }
      return result;
    }
  }, {
    key: "componentWillReceiveProps",
    value: function componentWillReceiveProps(nextProps) {
      if (this.state.value !== nextProps.value) {
        this.setState({ value: nextProps.value });
        var missing = this.isMissing(nextProps.value);
        var error = this.getError(nextProps.value);

        // Re-broadcast error and missing states up to the owner
        if (this.props.onErrorCountChange) {
          this.props.onErrorCountChange(this.props.name, error.validationError ? 1 : 0);
        }

        if (this.props.onMissingCountChange) {
          this.props.onMissingCountChange(this.props.name, missing ? 1 : 0);
        }
      }
    }
  }, {
    key: "componentDidMount",
    value: function componentDidMount() {
      var missing = this.isMissing(this.props.value);
      var error = this.getError(this.props.value);
      var value = this.props.value;

      this.setState({
        value: value,
        missing: missing,
        error: error.validationError,
        errorMsg: error.validationErrorMessage
      });

      // Initial error and missing states are fed up to the owner
      if (this.props.onErrorCountChange) {
        this.props.onErrorCountChange(this.props.name, error.validationError ? 1 : 0);
      }

      if (this.props.onMissingCountChange) {
        this.props.onMissingCountChange(this.props.name, missing ? 1 : 0);
      }
    }
  }, {
    key: "onBlur",
    value: function onBlur() {
      var value = this.refs.input.value;
      var missing = this.props.required && this.isEmpty(value);
      var error = this.getError(value);

      var cast = value;

      // State changes
      this.setState({
        value: value,
        missing: missing,
        error: error.validationError,
        errorMsg: error.validationErrorMessage
      });

      // Callbacks
      if (this.props.onChange) {
        if (_underscore2.default.has(this.props.rules, "type")) {
          switch (this.props.rules.type) {
            case "integer":
              cast = value === "" ? null : parseInt(value, 10);
              break;
            case "number":
              cast = value === "" ? null : parseFloat(value, 10);
              break;
            //pass
            default:
          }
        }
        this.props.onChange(this.props.name, cast);
      }
      if (this.props.onErrorCountChange) {
        this.props.onErrorCountChange(this.props.name, error.validationError ? 1 : 0);
      }
      if (this.props.onMissingCountChange) {
        this.props.onMissingCountChange(this.props.name, missing ? 1 : 0);
      }
    }
  }, {
    key: "onFocus",
    value: function onFocus() {
      this.setState({ error: false, errorMsg: "" });
    }
  }, {
    key: "inlineStyle",
    value: function inlineStyle(hasError, isMissing) {
      var color = "inherited";
      var background = "inherited";
      var borderLeftStyle = "inherited";
      var borderLeftColor = "inherited";
      var borderLeftWidth = 2;
      if (hasError) {
        color = "#b94a48";
        background = "#fff0f3";
        borderLeftStyle = "solid";
        borderLeftColor = "#b94a48";
      } else if (isMissing) {
        background = "floralwhite";
        borderLeftStyle = "solid";
        borderLeftColor = "orange";
      }
      return {
        color: color,
        background: background,
        borderLeftStyle: borderLeftStyle,
        borderLeftColor: borderLeftColor,
        borderLeftWidth: borderLeftWidth,
        height: 23,
        width: "100%",
        paddingLeft: 3
      };
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      if (this.props.edit) {
        // Error text
        var msg = this.state.error ? this.state.errorMsg : "";
        var helpClassName = "help-block";
        if (this.state.error) {
          helpClassName += " has-error";
        }

        // Warning style
        var style = this.isMissing(this.state.value) ? { background: "floralwhite" } : {};

        var key = (0, _stringHash2.default)(this.props.value || "");

        return _react2.default.createElement(
          "div",
          null,
          _react2.default.createElement("input", {
            required: true,
            key: key,
            ref: "input",
            className: "form-control input-sm",
            style: style,
            type: this.props.type,
            disabled: this.props.disabled,
            placeholder: this.props.placeholder,
            defaultValue: this.props.value,
            onBlur: function onBlur() {
              return _this2.onBlur();
            },
            onFocus: function onFocus() {
              return _this2.onFocus();
            }
          }),
          _react2.default.createElement(
            "div",
            { className: helpClassName },
            msg
          )
        );
      } else {
        var isMissing = this.isMissing(this.state.value);
        var hasError = this.state.error;
        var text = this.props.value;
        if (isMissing) {
          text = " ";
        }
        var _style = this.inlineStyle(hasError, isMissing);
        return _react2.default.createElement(
          "div",
          { style: _style },
          text
        );
      }
    }
  }]);

  return TextEdit;
}(_react2.default.Component);

exports.default = (0, _formGroup2.default)(TextEdit);