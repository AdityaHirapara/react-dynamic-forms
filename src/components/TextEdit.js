/**
 *  Copyright (c) 2015, The Regents of the University of California,
 *  through Lawrence Berkeley National Laboratory (subject to receipt
 *  of any required approvals from the U.S. Dept. of Energy).
 *  All rights reserved.
 *
 *  This source code is licensed under the BSD-style license found in the
 *  LICENSE file in the root directory of this source tree.
 */

import React from "react";
import { validate } from "revalidator";
import _ from "underscore";
import hash from "string-hash";

require("./textedit.css");

/**
 * Form control to edit a text field.
 * Set the initial value with 'initialValue' and set a callback for
 * value changed with 'onChange'.
 */
export default React.createClass({
  displayName: "TextEdit",
  getDefaultProps() {
    return { width: "100%" };
  },
  getInitialState() {
    return {
      initialValue: this.props.initialValue,
      value: this.props.initialValue,
      error: null,
      errorMsg: "",
      missing: false,
      type: "text"
    };
  },
  _isEmpty(value) {
    return _.isNull(value) || _.isUndefined(value) || value === "";
  },
  _isMissing(v) {
    return this.props.required && !this.props.disabled && this._isEmpty(v);
  },
  _getError(value) {
    const result = { validationError: false, validationErrorMessage: null };

    // If the user has a field blank then that is never an error. Likewise if the field
    // is disabled then that is never an error.
    if (this._isEmpty(value) || this.props.disabled) {
      return result;
    }

    // Validate the value with Revalidator, given the rules in this.props.rules
    let obj = {};
    obj[this.props.attr] = value;

    let attrValuePair = {};
    attrValuePair[this.props.attr] = this.props.rules;

    const rules = this.props.rules ? { properties: attrValuePair } : null;

    if (obj && rules) {
      const validation = validate(obj, rules, { cast: true });
      const name = this.props.name || "Value";

      let msg;
      if (!validation.valid) {
        msg = `${name} ${validation.errors[0].message}`;
        result.validationError = true;
        result.validationErrorMessage = msg;
      }
    }
    return result;
  },
  componentWillReceiveProps(nextProps) {
    if (this.state.initialValue !== nextProps.initialValue) {
      this.setState({
        initialValue: nextProps.initialValue,
        value: nextProps.initialValue
      });

      const missing = this._isMissing(nextProps.initialValue);
      const error = this._getError(nextProps.initialValue);

      // Re-broadcast error and missing states up to the owner
      if (this.props.onErrorCountChange) {
        this.props.onErrorCountChange(
          this.props.attr,
          error.validationError ? 1 : 0
        );
      }

      if (this.props.onMissingCountChange) {
        this.props.onMissingCountChange(this.props.attr, missing ? 1 : 0);
      }
    }
  },
  componentDidMount() {
    const missing = this._isMissing(this.props.initialValue);
    const error = this._getError(this.props.initialValue);
    const value = this.props.initialValue;

    this.setState({
      value,
      missing,
      error: error.validationError,
      errorMsg: error.validationErrorMessage
    });

    // Initial error and missing states are fed up to the owner
    if (this.props.onErrorCountChange) {
      this.props.onErrorCountChange(
        this.props.attr,
        error.validationError ? 1 : 0
      );
    }

    if (this.props.onMissingCountChange) {
      this.props.onMissingCountChange(this.props.attr, missing ? 1 : 0);
    }
  },
  onBlur() {
    const value = this.refs.input.value;
    const missing = this.props.required && this._isEmpty(value);
    const error = this._getError(value);

    let cast = value;

    // State changes
    this.setState({
      value,
      missing,
      error: error.validationError,
      errorMsg: error.validationErrorMessage
    });

    // Callbacks
    if (this.props.onChange) {
      if (_.has(this.props.rules, "type")) {
        switch (this.props.rules.type) {
          case "integer":
            cast = value === "" ? null : parseInt(value, 10);
            break;
          case "number":
            cast = value === "" ? null : parseFloat(value, 10);
            break;
          //pass
          default:
        }
      }
      this.props.onChange(this.props.attr, cast);
    }
    if (this.props.onErrorCountChange) {
      this.props.onErrorCountChange(
        this.props.attr,
        error.validationError ? 1 : 0
      );
    }
    if (this.props.onMissingCountChange) {
      this.props.onMissingCountChange(this.props.attr, missing ? 1 : 0);
    }
  },
  onFocus() {
    this.setState({ error: false, errorMsg: "" });
  },
  render() {
    let msg = "";
    const w = _.isUndefined(this.props.width) ? "100%" : this.props.width;
    const style = { width: w };

    let className = "";
    const requiredError = this.props.showRequired &&
      this._isMissing(this.state.value);
    if (this.state.error || requiredError) {
      className = "has-error";
    }

    if (this.state.error) {
      msg = this.state.errorMsg;
    }

    let helpClassName = "help-block";
    if (this.state.error) {
      helpClassName += " has-error";
    }

    const key = hash(this.state.initialValue || "");

    return (
      <div className={className}>
        <input
          required
          key={key}
          style={style}
          className="form-control input-sm"
          type={this.props.type}
          ref="input"
          disabled={this.props.disabled}
          placeholder={this.props.placeholder}
          defaultValue={this.state.value}
          onBlur={this.onBlur}
          onFocus={this.onFocus}
        />
        <div className={helpClassName}>{msg}</div>
      </div>
    );
  }
});

