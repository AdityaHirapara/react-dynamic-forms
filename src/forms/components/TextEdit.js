/**
 *  Copyright (c) 2015-2017, The Regents of the University of California,
 *  through Lawrence Berkeley National Laboratory (subject to receipt
 *  of any required approvals from the U.S. Dept. of Energy).
 *  All rights reserved.
 *
 *  This source code is licensed under the BSD-style license found in the
 *  LICENSE file in the root directory of this source tree.
 */

import React from "react";
import { validate } from "revalidator";
import _ from "underscore";

import formGroup from "../formGroup";
import "./css/textedit.css";

/**
 * Form control to edit a text field.
 * Set the initial value with 'initialValue' and set a callback for
 * value changed with 'onChange'.
 */
class TextEdit extends React.Component {
  isEmpty(value) {
    return _.isNull(value) || _.isUndefined(value) || value === "";
  }

  isMissing(v) {
    return this.props.required && !this.props.disabled && this.isEmpty(v);
  }

  getError(value) {
    const result = { validationError: false, validationErrorMessage: null };

    // If the user has a field blank then that is never an error. Likewise if the field
    // is disabled then that is never an error.
    if (this.isEmpty(value) || this.props.disabled) {
      return result;
    }

    // Validate the value with Revalidator, given the rules in this.props.rules
    let obj = {};
    obj[this.props.name] = value;

    let properties = {};
    properties[this.props.name] = this.props.validation;

    const rules = this.props.validation ? { properties } : null;
    if (obj && rules) {
      const validation = validate(obj, rules, { cast: true });
      const name = this.props.name || "Value";

      let msg;
      if (!validation.valid) {
        msg = `${name} ${validation.errors[0].message}`;
        result.validationError = true;
        result.validationErrorMessage = msg;
      }
    }
    return result;
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.value !== nextProps.value) {
      const missing = this.isMissing(nextProps.value);
      const { validationError } = this.getError(nextProps.value);

      // Broadcast error and missing states up to the owner
      if (this.props.onErrorCountChange) {
        this.props.onErrorCountChange(this.props.name, validationError ? 1 : 0);
      }

      if (this.props.onMissingCountChange) {
        this.props.onMissingCountChange(this.props.name, missing ? 1 : 0);
      }
    }
  }

  componentDidMount() {
    const missing = this.isMissing(this.props.value);
    const { validationError } = this.getError(this.props.value);

    // Initial error and missing states are fed up to the owner
    if (this.props.onErrorCountChange) {
      this.props.onErrorCountChange(this.props.name, validationError ? 1 : 0);
    }

    if (this.props.onMissingCountChange) {
      this.props.onMissingCountChange(this.props.name, missing ? 1 : 0);
    }
  }

  onBlur() {
    const value = this.refs.input.value;
    const missing = this.props.required && this.isEmpty(value);
    const { validationError } = this.getError(value);
    let cast = value;

    // Callbacks
    if (this.props.onChange) {
      if (_.has(this.props.rules, "type")) {
        switch (this.props.rules.type) {
          case "integer":
            cast = value === "" ? null : parseInt(value, 10);
            break;
          case "number":
            cast = value === "" ? null : parseFloat(value, 10);
            break;
          //pass
          default:
        }
      }
      this.props.onChange(this.props.name, cast);
    }
    if (this.props.onErrorCountChange) {
      this.props.onErrorCountChange(this.props.name, validationError ? 1 : 0);
    }
    if (this.props.onMissingCountChange) {
      this.props.onMissingCountChange(this.props.name, missing ? 1 : 0);
    }
  }

  inlineStyle(hasError, isMissing) {
    let color = "inherited";
    let background = "inherited";
    if (hasError) {
      color = "#b94a48";
      background = "#fff0f3";
    } else if (isMissing) {
      background = "floralwhite";
    }
    return {
      color,
      background,
      height: 23,
      width: "100%",
      paddingLeft: 3
    };
  }

  render() {
    // Control state
    const isMissing = this.isMissing(this.props.value);
    const { validationError, validationErrorMessage } = this.getError(
      this.props.value
    );

    if (this.props.edit) {
      // Error style/message
      let className = "";
      const msg = validationError ? validationErrorMessage : "";
      let helpClassName = "help-block";
      if (validationError) {
        helpClassName += " has-error";
        className = "has-error";
      }

      // Warning style
      const style = isMissing ? { background: "floralwhite" } : {};

      const type = this.props.type || "text";

      return (
        <div className={className}>
          <input
            key={this.props.value}
            ref="input"
            className="form-control input-sm"
            style={style}
            type={type}
            disabled={this.props.disabled}
            placeholder={this.props.placeholder}
            defaultValue={this.props.value}
            onBlur={() => this.onBlur()}
          />
          <div className={helpClassName}>{msg}</div>
        </div>
      );
    } else {
      const view = this.props.view;
      let text = this.props.value;
      if (isMissing) {
        text = " ";
      }
      const style = this.inlineStyle(validationError, isMissing);
      if (!view) {
        return <div style={style}>{text}</div>;
      } else {
        return view(text)
      }
    }
  }
}

export default formGroup(TextEdit);
