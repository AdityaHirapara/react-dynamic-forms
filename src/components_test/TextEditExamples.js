import TextEdit from "../components/TextEdit";
import React from "react";

export const Test1 = React.createClass({
    render() {
        return <TextEdit initialValue="Bob" width={300} />;
    }
});

