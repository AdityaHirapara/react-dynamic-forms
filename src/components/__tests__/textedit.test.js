/**
 *  Copyright (c) 2017, The Regents of the University of California,
 *  through Lawrence Berkeley National Laboratory (subject to receipt
 *  of any required approvals from the U.S. Dept. of Energy).
 *  All rights reserved.
 *
 *  This source code is licensed under the BSD-style license found in the
 *  LICENSE file in the root directory of this source tree.
 */

import renderer from "react-test-renderer";
import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import { Test1 } from "../../components_test/TextEditExamples";
import TextEdit from "../../components/TextEdit";

test("A very basic test of a TextEdit with an initial value and width", () => {
    const tree = renderer.create(<Test1 />).toJSON();
    expect(tree).toMatchSnapshot();
});

test("Interactivity of a TextEdit control", () => {
    const handleChange = jest.fn();

    const wrapper = mount(
        (
            <TextEdit
                attr="mytextedit"
                initialValue="bob"
                onChange={handleChange}
            />
        )
    );

    // Simulate user typing
    const input = wrapper.find("input");
    input.node.value = "fred";
    input.simulate("blur");

    // Get the result of the handleChange callback
    const [ attr, value ] = handleChange.mock.calls[0];
    expect(handleChange).toHaveBeenCalled();
    expect(attr).toBe("mytextedit");
    expect(value).toBe("fred");
});

test("TextEdit isRequired reporting", () => {
    const handleMissingCountChange = jest.fn();

    const wrapper = mount(
        (
            <TextEdit
                attr="mytextedit"
                initialValue="bob"
                required={true}
                onMissingCountChange={handleMissingCountChange}
            />
        )
    );

    // The TextEdit will evoke onMissingCountChange initially to report
    // up initial missing value counts, so handleMissingCountChange
    // should have been called once at this point, and since we provided
    // an initialValue the missing count should be 0
    const [ attr, missingBefore ] = handleMissingCountChange.mock.calls[0];
    expect(handleMissingCountChange.mock.calls.length).toBe(1);
    expect(attr).toBe("mytextedit");
    expect(missingBefore).toBe(0);

    // Simulate the user clearing the input field
    const input = wrapper.find("input");
    input.node.value = "";
    input.simulate("blur");

    // Get the result of the handleChange callback
    expect(handleMissingCountChange.mock.calls.length).toBe(2);
    const [ _, missingAfter ] = handleMissingCountChange.mock.calls[1];
    expect(missingAfter).toBe(1);
});

test("TextEdit isRequired snapshot test", () => {
    const handleMissingCountChange = jest.fn();

    const wrapper = mount(
        (
            <TextEdit
                attr="mytextedit"
                initialValue="bob"
                required={true}
                showRequired={true}
                onMissingCountChange={handleMissingCountChange}
            />
        )
    );
    const input = wrapper.find("input");
    const div = wrapper.find("div");

    // Get an initial snapshot of the TextEdit
    expect(toJson(div)).toMatchSnapshot();

    // Check if it shows an error (it shouldn't yet)
    expect(div.node.className).toBe("");

    // Simulate the user clearing the input field
    input.node.value = "";
    input.simulate("blur");

    // Check if it shows an error now
    expect(div.node.className).toBe("has-error");

    // Get an after snapshot of the TextEdit
    expect(toJson(div)).toMatchSnapshot();
});

