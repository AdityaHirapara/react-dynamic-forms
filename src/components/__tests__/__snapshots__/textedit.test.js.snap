exports[`test A very basic test of a TextEdit with an initial value and width 1`] = `
<div
  className="">
  <input
    className="form-control input-sm"
    defaultValue="Bob"
    disabled={undefined}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder={undefined}
    required={true}
    style={
      Object {
        "width": 300,
      }
    }
    type="text" />
  <div
    className="help-block">
    
  </div>
</div>
`;

exports[`test TextEdit isRequired snapshot test 1`] = `
<div
  className="">
  <input
    className="form-control input-sm"
    defaultValue="bob"
    onBlur={[Function]}
    onFocus={[Function]}
    required={true}
    style={
      Object {
        "width": "100%",
      }
    }
    type="text" />
  <div
    className="help-block" />
</div>
`;

exports[`test TextEdit isRequired snapshot test 2`] = `
<div
  className="has-error">
  <input
    className="form-control input-sm"
    defaultValue=""
    onBlur={[Function]}
    onFocus={[Function]}
    required={true}
    style={
      Object {
        "width": "100%",
      }
    }
    type="text" />
  <div
    className="help-block" />
</div>
`;
